### DISCUSSION TITLE
Why isn't Callable an Object?

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#6974] [Scripting]
I can imagine extending Callable and adding some custom fields to it, storing where some data for example.
Will it be heavy if the Callable were an Object

**Comment by AThousandShips** [comment]
It will increase the pressure on `ObjectID`s by orders of magnitude, don't think this is a wise choice

**Reply by AThousandShips** [reply]
There's room for a total of `2^24-1` objects at any one time, that's `16777215`, a lot of course, but the number of callables in the engine easily outnumber the number of objects, one for each connection

For example:
* Each `CanvasItem` has at least one connection active if it is under any viewport, and an additional one for each `Control`
* As does most nodes that use resources, like `MeshInstance`

**Reply by Calinou** [reply]
Object also requires more heap allocation. If Callable was an Object, it would have even more overhead.

**Reply by AThousandShips** [reply]
Oh that's true, relevant in `Variant` where it is stored in the fixed part of the data
